snap-core/src/main/java/org/esa/snap/core/dataio/dimap/CacheBlock.java

```diff
@@ -0,0 +1,63 @@
+package org.esa.snap.core.dataio.dimap;
+
+import org.esa.snap.core.datamodel.ProductData;
+
+import java.awt.Rectangle;
+import java.awt.geom.Area;
+
+class CacheBlock {
+
+    private final int yStart;
+    private final int width;
+    private final int height;
+    private final Area unwrittenSpace;
+
+    private ProductData bufferData;
+
+    CacheBlock(int yStart, int width, int height, int dataType, double noDataValue) {
```

Comment:
Why only yStart and no xstart?

Reply:
Because the cache always cover the full product width and therefore always starts at x=0

Reply:
Yes, okay. I was just confused because the produt also starts always a y=0, but maybe not the cache.
Anyway, not a real issue.